app/src/main/java/com/alexbar/layoutbasic/screen/confirm_payment/widgets/AddNewCreditCardDialog.kt

```diff
@@ -0,0 +1,169 @@
+package com.alexbar.layoutbasic.screen.confirm_payment.widgets
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.shadow
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.alexbar.layoutbasic.ui.theme.Dimens.dimen_16_dp
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_cancel
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_full_name
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_month
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_number
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_save
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_year
+
+@Composable
+fun AddCreditCardDialog(
```

Comment:
general suggestion : Separate Functions for UI Elements
inside this file you can create another comp fun like :
```
@Composable
fun CreditCardDialogUI(
    name: String,
    number: String,
    expirationMonth: String,
    expirationYear: String,
    onNameChange: (String) -> Unit,
    onNumberChange: (String) -> Unit,
    onExpirationMonthChange: (String) -> Unit,
    onExpirationYearChange: (String) -> Unit,
    onCreateCard: () -> Unit,
    onDismissRequest: () -> Unit,
) {
...
}
```

and then you can use this func into your main fun...to have more readable code :

```
@Composable
fun AddCreditCardDialog(
    onCreateCard: (CreditCardInfo) -> Unit,
    onDismissRequest: () -> Unit,
) {
    var name by remember { mutableStateOf("") }
    var number by remember { mutableStateOf("") }
    var expirationMonth by remember { mutableStateOf("") }
    var expirationYear by remember { mutableStateOf("") }

    Dialog(onDismissRequest = onDismissRequest) {
        CreditCardDialogUI(
            name = name,
            number = number,
            expirationMonth = expirationMonth,
            expirationYear = expirationYear,
            onNameChange = { name = it },
            onNumberChange = { number = it.take(16) },
            onExpirationMonthChange = { expirationMonth = it },
            onExpirationYearChange = { expirationYear = it },
            onCreateCard = {
                onCreateCard(
                    CreditCardInfo(
                        name,
                        number,
                        expirationMonth,
                        expirationYear
                    )
                )
                onDismissRequest()
            },
            onDismissRequest = onDismissRequest
        )
    }
}
```

These changes aim to enhance code readability and maintainability by breaking down the complex function into smaller